source/common/upstream/eds.cc

```diff
@@ -47,11 +47,13 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
     const std::string& zone = locality_lb_endpoint.locality().zone();
 
     for (const auto& lb_endpoint : locality_lb_endpoint.lb_endpoints()) {
+      const bool canary =
+          Config::Utility::metadataValue(lb_endpoint.metadata(), "envoy.lb", "canary").bool_value();
```

Comment:
Can we add "envoy.lb" and "canary" to some type of MetadataConstants class or something? We will be referencing well known metadata in various places in the code and it would be good to put the well known things in a central place.